servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java

```diff
@@ -996,9 +987,9 @@ private static Entry maybeTruncateToDepth(Entry entry, int depth) {
    *
    * @param filter The filter to filter by
    */
-  protected Predicate<KeyEntry> filterEntries(String filter) {
+  protected BiPredicate<ContentKey, Content.Type> filterEntries(String filter) {
     if (Strings.isNullOrEmpty(filter)) {
-      return x -> true;
+      return null;
```

Comment:
I suggested this change initially but I'm not sure anymore it buys us any good. Reverting this change will simplify lines 910-916.

Reply:
Maybe add a utility method for combining nullable filters?

Reply:
i think its helpful because it can avoid determining the contenttype if no predicate is present (instead of an always true predicate)

Reply:
> Maybe add a utility method for combining nullable filters?

added... not sure if its much nicer :sweat_smile:

Reply:
thx!